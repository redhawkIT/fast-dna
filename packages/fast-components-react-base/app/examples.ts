import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };

import CheckboxExamples from "../src/checkbox/examples.data";
export { CheckboxExamples };

import DividerExamples from "../src/divider/examples.data";
export { DividerExamples };

import ImageExamples from "../src/image/examples.data";
export { ImageExamples };

import LabelExamples from "../src/label/examples.data";
export {LabelExamples};

import ToggleExamples from "../src/toggle/examples.data";
export { ToggleExamples };

import TypographyExamples from "../src/typography/examples.data";
export { TypographyExamples };

import HypertextExamples from "../src/hypertext/examples.data";
export { HypertextExamples };
