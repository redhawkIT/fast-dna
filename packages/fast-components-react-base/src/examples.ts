import * as React from "react";

export type IExampleData<T> = T;

export interface IGenericExample<T> {
    name: string;

    component: React.ComponentClass;

    data: Array<IExampleData<T>>;
}
