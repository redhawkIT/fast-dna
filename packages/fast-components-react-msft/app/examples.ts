import ButtonExample from "../src/button/examples.data";
export {ButtonExample};

import DividerExample from "../src/divider/examples.data";
export {DividerExample};

import ToggleExamples from "../src/toggle/examples.data";
export {ToggleExamples};
