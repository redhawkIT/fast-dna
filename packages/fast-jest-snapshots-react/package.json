{
  "name": "@microsoft/fast-jest-snapshots-react",
  "description": "A small utility for automating Jest snapshot creation of React components",
  "version": "1.6.0",
  "author": {
    "name": "Microsoft",
    "email": "fwq@microsoft.com"
  },
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Microsoft/fast-dna.git"
  },
  "private": true,
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
    "prepare": "tsc -p ./tsconfig.json",
    "test": "npm run tslint",
    "start": "webpack-dev-server",
    "tslint": "tslint -c ./tslint.json '**/*.ts{,x}'",
    "tslint:fix": "tslint -c ./tslint.json --fix '**/*.ts{,x}'"
  },
  "devDependencies": {
    "@microsoft/fast-tslint-rules": "^1.6.0",
    "@types/jest": "^22.2.2",
    "jest": "^22.4.3",
    "react": "^16.3.0",
    "ts-jest": "^22.4.2",
    "tslint": "^5.9.1",
    "typescript": "^2.8.1"
  },
  "dependencies": {
    "@types/react": "^16.3.0",
    "react-test-renderer": "^16.3.0"
  },
  "peerDependencies": {
    "jest": "^22.0.0",
    "react": "^16.3.0"
  }
}
