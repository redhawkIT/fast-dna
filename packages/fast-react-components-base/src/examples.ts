import * as React from "react";

export type IExampleData<T> = T;

export interface IGenericExample<T> {
    name: string;

    component: React.ComponentClass;

    data: IExampleData<T>[];
}