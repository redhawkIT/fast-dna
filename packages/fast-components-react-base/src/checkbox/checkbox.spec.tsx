import * as React from "react";
import * as ShallowRenderer from "react-test-renderer/shallow";
import * as Adapter from "enzyme-adapter-react-16";
import { configure, shallow } from "enzyme";
import examples from "./examples.data";
import { generateSnapshots } from "@microsoft/fast-jest-snapshots-react";
import {
    CheckboxProps,
    ICheckboxClassNameContract,
    ICheckboxHandledProps,
    ICheckboxManagedClasses,
    ICheckboxState,
    ICheckboxUnhandledProps
} from "./checkbox";

/*
 * Configure Enzyme
 */
configure({adapter: new Adapter()});

describe("checkbox snapshot", (): void => {
    generateSnapshots(examples);
});

describe("checkbox", (): void => {
    const Component: React.ComponentClass<ICheckboxHandledProps & ICheckboxManagedClasses> = examples.component;
    const managedClasses: ICheckboxClassNameContract = {
        checkbox: "checkbox-class",
        checkbox_disabled: "disabled-class",
        checkbox_input: "input-class",
        checkbox_label: "label-class",
        checkbox_span: "span-class",
    };

    test("should return an object that includes all valid props which are not enumerated as handledProps", () => {
        const handledProps: ICheckboxHandledProps & ICheckboxManagedClasses = {
            managedClasses
        };

        const unhandledProps: ICheckboxUnhandledProps = {
            "aria-hidden": true
        };

        const props: CheckboxProps = {...handledProps, ...unhandledProps};

        const rendered: any = shallow(
            <Component {...props} />
        );

        expect(rendered.prop("aria-hidden")).not.toBe(undefined);
        expect(rendered.prop("aria-hidden")).toEqual(true);
    });

    test("should correctly manage the disabled prop by adding a class and passing `disabled` to the input element", () => {
        const rendered: any = shallow(
            <Component managedClasses={managedClasses} disabled={true} />
        );

        expect(rendered.hasClass("disabled-class")).toBe(true);
        expect(rendered.find(".input-class[disabled]")).not.toBe(undefined);
        expect(rendered.find(".input-class[disabled]").prop("disabled")).toBe(true);
    });

    test("should correctly operate as an uncontrolled component and set initial state if `checked` prop is not passed", () => {
        const rendered: any = shallow(
            <Component managedClasses={managedClasses} />
        );

        const state: any = rendered.state("checked");

        expect(state).toBe(false);
    });

    test("should correctly operate as an uncontrolled component and handle `onChange` events when `onChange` prop is not passed", () => {
        const rendered: any = shallow(
            <Component managedClasses={managedClasses} />
        );

        expect(rendered.state("checked")).toBe(false);

        const input: any = rendered.find(".input-class");
        input.prop("onChange")(); // we have to fire the internal onChange from the input as we aren't passing one

        expect(rendered.state("checked")).toBe(true);
    });

    test("should correctly operate as a controlled component when `onChange` and `checked` prop is passed", () => {
        const onChange: any = jest.fn();
        const rendered: any = shallow(
            <Component managedClasses={managedClasses} checked={true} onChange={onChange} />
        );
        const input: any = rendered.find("input");

        input.simulate("change", { checked: false });

        expect(onChange).toHaveBeenCalled();
    });

    test("should correctly update the checked state if an updated `checked` prop is passed which differs from the state", () => {
        const rendered: any = shallow(
            <Component managedClasses={managedClasses} />
        );

        const newProps: Partial<ICheckboxHandledProps> = {
            checked: true
        };

        expect(rendered.state("checked")).toEqual(false);

        rendered.setProps(newProps);

        expect(rendered.state("checked")).toEqual(true);
    });
});
