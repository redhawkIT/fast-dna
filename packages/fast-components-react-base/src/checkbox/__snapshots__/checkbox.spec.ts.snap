// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox checkbox: 0 1`] = `
<label
  className="checkbox"
>
  <input
    checked={undefined}
    className="checkbox-input"
    disabled={undefined}
    onChange={undefined}
    type="checkbox"
  />
  <span
    className="checkbox-label"
  />
</label>
`;

exports[`checkbox checkbox: 1 1`] = `
<label
  className="checkbox"
>
  <input
    checked={undefined}
    className="checkbox-input"
    disabled={undefined}
    onChange={undefined}
    type="checkbox"
  />
  <span
    className="checkbox-label"
  />
</label>
`;

exports[`checkbox checkbox: 2 1`] = `
<div
  className="checkbox"
>
  <input
    checked={undefined}
    className="checkbox-input"
    disabled={undefined}
    onChange={undefined}
    type="checkbox"
  />
  <span
    className="checkbox-label"
  />
</div>
`;

exports[`checkbox checkbox: 3 1`] = `
<label
  className="checkbox"
>
  <input
    checked={undefined}
    className="checkbox-input"
    disabled={undefined}
    onChange={undefined}
    type="checkbox"
  />
  <span
    className="checkbox-label"
  />
</label>
`;
